### PR TITLE
K8s YouTube vanity URL

### PR DESCRIPTION
**Pull request by chris-short**
First off, was trying to rebase #450 but, I failed horribly so my apologies for the delay in getting this done AND for screwing up the previous PR.

Hello! Every week I typically have to figure out one URL or another to get people to the YouTube channel. Creating this redirect would make my life easier. The @kubernetes/youtube-admins team might agree too.

NOTE: This is for the top-level Kubernetes YouTube channel; not any individual recording/stream/etc.

Signed-off-by: Chris Short <chris@chrisshort.net>

### CODE REVIEW

**Comment by cblecker** [comment]
/approve

I'm good with this. Assigning to Tim as I think he's the only one with the keys right now to deploy this.

/assign @thockin

**Comment by stp-ip** [comment]
Looks good.
Just needs Tim to push the DNS changes and reload the nginx config.
/approve

File: dns/zone-configs/kubernetes.io.yaml [hunk 1]
```diff
@@ -183,3 +183,6 @@ pfjqlwt67ty3.testgrid:
 velodrome:
   type: CNAME
   value: velodrome.k8s.io.
+youtube:
```
Comment (thockin) [inline dns/zone-configs/kubernetes.io.yaml]: Why yt.k8s.io and not yt.kubernetes.io ?

Comment (thockin) [reply dns/zone-configs/kubernetes.io.yaml]: and why not go.k8s.io/youtube ?

Comment (chris-short) [reply dns/zone-configs/kubernetes.io.yaml]: The answer to both is fewer characters, easier to remember. I can add them if you'd like but, I don't know if they'd be utilized in lieu of the others.

**Comment by thockin** [comment]
wrt k8s/kubernetes - it is simpler overall to have most names work in both,
so we don't question why it is different.

I don't feel strongly about go/ vs a full name for this case

**Comment by chris-short** [comment]
Consistency makes sense. Will fix up and push changes shortly.

**Comment by chris-short** [comment]
Alright, @thockin. I _think_ this covers all the cases.

File: k8s.io/configmap-nginx.yaml [hunk 2]
```diff
@@ -244,6 +244,8 @@ data:
           rewrite ^/test-health$     http://velodrome.k8s.io/dashboard/db/bigquery-metrics redirect;
           rewrite ^/test-history$    https://storage.googleapis.com/kubernetes-test-history/static/index.html redirect;
           rewrite ^/triage$          https://storage.googleapis.com/k8s-gubernator/triage/index.html redirect;
+          rewrite ^/youtube$          https://www.youtube.com/c/kubernetescommunity redirect;
```
Comment (thockin) [inline k8s.io/configmap-nginx.yaml]: Sorry, I wasn't clear - we don't need a DNS name *and* a go link.  One or the other :)

**Comment by chris-short** [comment]
Oh Tim...

Good?

File: k8s.io/README.md [hunk 3]
```diff
@@ -52,6 +53,8 @@ Redirections
 - https://go.k8s.io/test-health
 - https://go.k8s.io/test-history
 - https://go.k8s.io/triage
+- https://go.k8s.io/youtube
```
Comment (thockin) [inline k8s.io/README.md]: Forgot these?

**Comment by thockin** [comment]
so....close....

:)

File: k8s.io/README.md (hunk 3 above)
Comment (chris-short) [reply k8s.io/README.md]: dag nabbit

**Comment by chris-short** [comment]
Maybe?

Maybe

**Comment by chris-short** [comment]
@thockin bump

File: k8s.io/configmap-nginx.yaml [hunk 4]
```diff
@@ -287,6 +287,12 @@ data:
 
         location / {
           rewrite ^/.*$  https://prow.k8s.io/tide redirect;
+
+      server {
+        server_name yt.k8s.io youtube.k8s.io youtube.kubernetes.io yt.kubernetes.io;
```
Comment (thockin) [inline k8s.io/configmap-nginx.yaml]: We need to add these to the certificate, too.  Followup is OK.

Comment (stp-ip) [reply k8s.io/configmap-nginx.yaml]: pointer: https://github.com/kubernetes/k8s.io/blob/master/k8s.io/certificate-prod.yaml

**Comment by thockin** [comment]
I'll send the followup as I am deploying this change

**Comment by thockin** [comment]
https://github.com/jetstack/cert-manager/issues/2005  @munnerz

**Pull request merged**